src/Workspaces/Core/Portable/FindSymbols/Declarations/DeclarationFinder_AllDeclarations.cs

```diff
@@ -93,19 +93,22 @@ internal static partial class DeclarationFinder
         private static async Task<(bool, ImmutableArray<SymbolAndProjectId>)> TryFindAllDeclarationsWithNormalQueryInRemoteProcessAsync(
             Project project, SearchQuery query, SymbolFilter criteria, CancellationToken cancellationToken)
         {
-            using (var session = await SymbolFinder.TryGetRemoteSessionAsync(
-                project.Solution, cancellationToken).ConfigureAwait(false))
+            if (RemoteSupportedLanguages.IsSupported(project.Language))
```

Comment:
this check seems potentially appropriate.  we should comment that this can be called through something like SolutionExplorer and that if this is not C#/VB that the search really can't happen at all (either on the host or remote side).